pkg/server/config.go

```diff
@@ -87,6 +88,7 @@ type ExtraConfig struct {
 	// clients
 	DynamicClusterClient       dynamic.ClusterInterface
 	KubeClusterClient          kubernetes.ClusterInterface
+	DeepSARClient              kubernetes.ClusterInterface
```

Comment:
(no action required, could be a follow-up if needed) I wonder if we should make this a different type, such as
```go
type DeepSARClient struct {
  kubernetes.ClusterInterface
}
```

so that you can't inject any arbitrary `kubernetes.ClusterInterface` into an admission plugin that needs a deep SAR client.

Reply:
yeah, that's often the crux with dependency injection. It is still the same type, implements the same interface. i am not sure that changing the type is the right answer though, opnions @stevekuznetsov ?

Reply:
This should be `kubernetes.Interface` though

Reply:
@varshaprasad96 you're right, my bad

Reply:
Np, https://github.com/kcp-dev/kcp/pull/1753 should fix it